dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -226,6 +227,14 @@ private String buildTrackedEntityInstanceHql( TrackedEntityInstanceQueryParams p
             hql += hlp.whereAnd() + "tei.lastUpdated > tei.lastSynchronized";
         }
 
+        //Going for comparing milliseconds instead of always creating new Date( 0 );
+        if ( params.getSkipChangedBefore() != null && params.getSkipChangedBefore().getTime() > 0 )

```

Comment:
isnt getTime always > 0 ?

Reply:
If user did not initilize it itself, then it is set to Date(0) by default.